src/cortexpy/tesserae.py

```diff
@@ -136,8 +159,19 @@ def __align_all(self, panel):
         lsize_l = np.log(size_l)
 
         max_r, pos_max, state_max, who_max = self.__initialization(query, panel, lsize_l)
-        max_r, pos_max, state_max, who_max = self.__recurrence(query, panel, lsize_l, l1, max_r, pos_max, state_max, who_max)
-        cp = self.__termination(l1, pos_max, state_max, who_max)
+        max_r, pos_max, state_max, who_max = self.__recurrence(query, panel, lsize_l, l1, max_r, pos_max, state_max, who_max, store_states=self.mem_limit)
+        cp, pos_max, who_max, state_max = self.__termination(np.remainder(self.qlen, self.traceback_limit) if self.mem_limit else l1, pos_max, state_max, who_max, 2*self.maxl)
```

Comment:
All these return values are getting a bit much. Can we put them in an object and just pass that object around?